packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -1784,15 +1786,28 @@ All @auth directives used on field definitions are performed when the field is r
     ctx.mapResourceToStack(parent.name.value, resolverResourceId);
   }
 
-  private addSubscriptionOwnerArgument(ctx: TransformerContext, resolver: Resolver, ownerRules: AuthRule[], makeNonNull: boolean = false) {
+  private addSubscriptionOwnerArgument(
+    ctx: TransformerContext,
+    resolver: Resolver,
+    currentFields: any,
+    ownerRules: AuthRule[],
+    makeNonNull: boolean = false
+  ) {
     let subscription = ctx.getSubscription();
     let createField: FieldDefinitionNode = subscription.fields.find(
       field => field.name.value === resolver.Properties.FieldName
     ) as FieldDefinitionNode;
-    const nameNode: any = makeNonNull ? makeNonNullType(makeNamedType('String')) : makeNamedType('String');
-    // const createArguments = [makeInputValueDefinition(DEFAULT_OWNER_FIELD, nameNode)];
     const ownerArgumentList = ownerRules.map(rule => {
-      return makeInputValueDefinition(rule.ownerField || DEFAULT_OWNER_FIELD, nameNode);
+      let argNode: any;
+      // if ownerField is defined read from fields if the value exists
+      if (rule.ownerField) {
+        argNode = currentFields[rule.ownerField] ? unwrapNonNull(currentFields[rule.ownerField]) : makeNamedType('String');
```

Comment:
when would this be false?

Reply:
This happens in the event that a ownerField rule is defined and the field is not added such as adding an ownerField named "notAllowed".